gammapy/makers/utils.py

```diff
@@ -422,3 +424,161 @@ def make_theta_squared_table(
     table.meta["ON_RA"] = position.icrs.ra
     table.meta["ON_DEC"] = position.icrs.dec
     return table
+
+
+def get_rad_max_vs_energy(rad_max, pointing, geom):
+    """Obtain the values of `RAD_MAX` at a given offset and for an array of
+    estimated energy values (in the geom energy axis).
+
+    Parameters
+    ----------
+    rad_max : `~gammapy.irf.RadMax2D`
+        the RAD_MAX_2D table IRF
+    geom : `~gammapy.maps.Geom`
+        the map geom to be used
+    pointing : `~astropy.coordinates.SkyCoord`
+        pointing direction
+
+    Returns
+    -------
+    array : `~astropy.units.Quantity`
+        Values of the `RAD_MAX` corresponding to each estimated energy bin center.
+    """
+    on_center = geom.center_skydir
+    offset = on_center.separation(pointing)
+
+    rad_max_vals = rad_max.evaluate(
+        offset=offset, energy=geom.axes["energy"].center
+    )
+
+    return rad_max_vals
+
+
+def make_counts_rad_max(geom, rad_max, events):
+    """Extract the counts using for the ON region size the values in the
+    `RAD_MAX_2D` table.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        reference map geom
+    rad_max : `~gammapy.irf.RadMax2D`
+        the RAD_MAX_2D table IRF
+    events : `~gammapy.data.EventList`
+        event list to be used to compute the ON counts
+
+    Returns
+    -------
+    counts : `~gammapy.maps.RegionNDMap`
+        Counts vs estimated energy extracted from the ON region.
+    """
+    rad_max = get_rad_max_vs_energy(rad_max, events.pointing_radec, geom)
+
+    counts_list = []
+
+    # create and fill a map per each energy bin, fetch the counts
+    for i, rad in enumerate(rad_max):
```

Comment:
Maybe a comment here:
since RAD_MAX is determined independently of any projection, the region based selection is not a necessity. It might be more efficient here to perform the filtering on the `EventList` directly.

Reply:
Yes, just using `coordinate.separation(source_position) < rad_max` should be more efficient

Reply:
But the `RegionGeom` serves also the purpose of filtering by energy, if you look at it, it is defined by a single energy bin
If I filter your way then I have to filter also by energy, right?